traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.DNS.tpl.html

```diff
@@ -228,6 +228,22 @@ <h3>Current Value</h3>
                             </aside>
                         </div>
                     </div>
+                    <div class="form-group" ng-class="{'has-error': hasError(generalConfig.topology), 'has-feedback': hasError(generalConfig.topology)}">
+                        <label class="has-tooltip control-label col-md-2 col-sm-2 col-xs-12" for="topology">Topology<div class="helptooltip">
+                            <div class="helptext">A topology is used to determine the caching layers and path to a delivery service's origin.</div>
+                        </div>
+                        </label>
+                        <div class="col-md-10 col-sm-10 col-xs-12">
+                            <select id="topology" name="topology" class="form-control" ng-model="deliveryService.topology" ng-options="topology.name as topology.name for topology in topologies">
+                                <option selected value="">None</option>
+                            </select>
+                            <small ng-show="deliveryService.topology"><a href="/#!/topologies/edit?name={{deliveryService.topology}}" target="_blank">View Details&nbsp;&nbsp;<i class="fa fs-xs fa-external-link"></i></a></small>
```

Comment:
If you're putting a data-bound value into an `href` you should be using [`ngHref`](https://docs.angularjs.org/api/ng/directive/ngHref#!). You'll probably also want the entire path to which it re-directs to be in the model, as that will allow the routing to change location without reloading the page - the docs have an example of all the different ways you can use it.

Reply:
yeah, should probably be an ngHref for all of the `View Details` links across all of the forms but i think i'll create a tech debt global issue for that.

Reply:
https://github.com/apache/trafficcontrol/issues/4719